src/components/OrderBookWidget.tsx

```diff
@@ -224,103 +223,257 @@ function _printOrderBook(pricePoints: PricePointDetails[], baseToken: TokenDetai
+    xAxis.strictMinMax = true
+    // How small we want the column separators be, in pixels
+    // https://www.amcharts.com/docs/v4/reference/axisrendererx/#minGridDistance_property
+    xAxis.renderer.minGridDistance = 40
+
+    // To start zoomed in when data loads
+    // https://www.amcharts.com/docs/v4/reference/valueaxis/#keepSelection_property
+    xAxis.keepSelection = true
+    // To allow we to zoom reaaaaly tiny fractions
+    xAxis.maxZoomFactor = 10 ** 18
+
+    const yAxis = _chart.yAxes.push(new am4charts.ValueAxis())
+
+    yAxis.keepSelection = true
+    yAxis.maxZoomFactor = 10 ** 18
+
+    // Colors
+    const colors = {
+      green: '#3d7542',
+      red: '#dc1235',
+    }
+
+    // Create series
+    const bidSeries = _chart.series.push(new am4charts.StepLineSeries())
+    bidSeries.dataFields.valueX = 'priceNumber'
+    bidSeries.dataFields.valueY = 'bidValueY'
+    bidSeries.strokeWidth = 1
+    bidSeries.stroke = am4core.color(colors.green)
+    bidSeries.fill = bidSeries.stroke
+    bidSeries.fillOpacity = 0.1
+
+    const askSeries = _chart.series.push(new am4charts.StepLineSeries())
+    askSeries.dataFields.valueX = 'priceNumber'
+    askSeries.dataFields.valueY = 'askValueY'
+    askSeries.strokeWidth = 1
+    askSeries.stroke = am4core.color(colors.red)
+    askSeries.fill = askSeries.stroke
+    askSeries.fillOpacity = 0.1
+
+    // Add cursor
+    _chart.cursor = new am4charts.XYCursor()
+
+    _chart.cursor.snapToSeries = [bidSeries, askSeries]
+
+    am4core.useTheme(am4themesSpiritedaway)
+    am4core.options.autoSetClassName = true
+
+    _chart.scrollbarX = new am4core.Scrollbar()
+    _chart.scrollbarY = new am4core.Scrollbar()
 
-export default OrderBookWidget
+    setChart(_chart)
+
+    return (): void => _chart.dispose()
+    // We'll create only one instance as long as the component is not unmounted
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (chart) {
+      const baseTokenLabel = safeTokenName(baseToken)
+      const quoteTokenLabel = safeTokenName(quoteToken)
+      const market = baseTokenLabel + '-' + quoteTokenLabel
+
+      const networkDescription = networkId !== Network.Mainnet ? `${getNetworkFromId(networkId)} ` : ''
+
+      // Axes configs
+      const xAxis = chart.xAxes.values[0] as am4charts.ValueAxis<am4charts.AxisRenderer>
+      xAxis.title.text = `${networkDescription} Price (${quoteTokenLabel})`
+
+      const yAxis = chart.yAxes.values[0] as am4charts.ValueAxis<am4charts.AxisRenderer>
+      yAxis.title.text = baseTokenLabel
+
+      // Add data
+      chart.dataSource.url = dexPriceEstimatorApi.getOrderBookUrl({
+        baseTokenId: baseToken.id,
+        quoteTokenId: quoteToken.id,
+        hops,
+        networkId,
+      })
+
+      // Removing any previous event handler
+      chart.dataSource.adapter.remove('parsedData')
+      // Adding new event handler
+      chart.dataSource.adapter.add('parsedData', data => {
```

Comment:
nit, would it be worth it to move `parsedData` to another function. This `iseEffect` is too big. It would be good to break it down into smaller parts for readability

Reply:
Did some refactoring, still more to do.